### DISCUSSION TITLE
Cyberduck for Mac Fails To Save Information In Keychain

### DISCUSSION THREAD

**Discussion by scrutinizer11** [iterate-ch/cyberduck#14539] [Q&A]
The issue appeared out of nowhere. Every time I start Cyberduck trying to connect to services, it requires the Application Key ID. The Console shows the error (the example below is the error output pointing to a failure to use the saved credentials for Backblaze)

`17/04/2023 03:05:50.735 Cyberduck: [main] ERROR ch.cyberduck.ui.cocoa.controller.DefaultBookmarkController - Failure saving credentials for Host{credentials=Credentials{user='[redacted]', token='', identity=null}, hostname='api.backblaze.com', defaultpath='/agate0pcthman0wallow0unscrewed', port=443, protocol=Profile{parent=b2, vendor=iterate GmbH, description=Backblaze B2 Cloud Storage, image=Local{path='/var/folders/3q/k7smvfkn33z49q4xdy_lfv7h0000gn/T/6b5e3974-7cdd-46a7-b141-8a3c0d43dbc9/Q1iYly30'}}} in keychain. Failure saving credentials for 00522a27b0a69150000000001 in Keychain. Please verify disk permissions.
`

When I enter that key, it connects only as long as its window is opened or the CB is running.

The application sits in **~/Applications** (not **/Applications**) but I'm not confident it makes a difference. At the GUI level, the Console message above corresponds to a suggestion to verify Disk permissions which didn't have any effect. And where is CB's keychain on my Mac?

**Comment by dkocher** [comment]
You may want to look at the output of 

```
log stream --predicate '(process =="Cyberduck")' --level error
```

in Terminal.app if it has any indication about the cause of the problem accessing the keychain.
